internal/bminventory/inventory.go

```diff
@@ -250,11 +250,27 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 func (b *bareMetalInventory) GetDiscoveryIgnition(ctx context.Context, params installer.GetDiscoveryIgnitionParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
 
+	cluster, err := common.GetClusterFromDB(b.db, params.ClusterID, common.SkipEagerLoading)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get cluster: %s", params.ClusterID)
+		return common.GenerateErrorResponder(err)
+	}
+
 	infraEnv, err := common.GetInfraEnvFromDB(b.db, params.ClusterID)
 	if err != nil {
+		log.WithError(err).Errorf("failed to get infra-env: %s", params.ClusterID)
 		return common.GenerateErrorResponder(err)
 	}
 
+	// update pull secret and proxy, since they could have been set during cluster update
+	proxy := models.Proxy{
```

Comment:
This means that we ignore these parameters in the infraEnv and take it from the cluster even for v2.  Is this the intention ?